testproject/testapp/tests.py

```diff
@@ -235,6 +241,10 @@ def test_unconditional_delete(self):
         with utils.CaptureQueriesContext(connections['manticore']) as ctx:
             self.model.objects.all().delete()
         sql = ctx.captured_queries[-1]['sql']
+
+        if django.VERSION >= (4, 2):
+            sql = ctx.captured_queries[-2]['sql']
```

Comment:
need a comment, what is the last query after `delete()` call

Reply:
Added a comment

"# In django 4.2 the last query is "COMMIT""

Reply:
Well, default django TestCase doesn't allow COMMIT here. Transaction management in tests is broken?

Reply:
Yes, transactions are disabled in tests because one of the databases does not support them.

https://github.com/just-work/django-manticore/blob/d4fc4fcf2e7f83bbef77ac28d0b64c197f047f4e/manticore/backend/base.py#L62

Reply:
@tumb1er In Django 4.2, logging of management queries was added, so they appeared here. https://docs.djangoproject.com/en/4.2/releases/4.2/#logging

Reply:
If transaction management is disabled, why "COMMIT" is here?

In my experience, there are a lot of feature-flags in DatabaseWrapper that changes django orm behavior. This flags are not documented and not a part of public API. So, my guess is that some flag is missing now for manticore backend and ORM is trying to commit something while connection is in autocommit mode. 

Additionally, I would try to write same test for "normal" database connection (pg/my/sqlite) to compare queries between rdbms and manticore backends.

Reply:
Because transaction support is disabled, so all changes are recorded. And that's why we see COMMIT here. In previous versions of Django we did not see this only because it was not logged.

Reply:
If transaction support is disabled, so connection should be in autocommit mode. I need more details to decide what's next here.

Reply:
Here `ctx.captured_queries` for django < 4.2 looks like:
```[{'sql': 'TRUNCATE RTINDEX `cluster`:`test___testapp_testmodel`', 'time': '0.000'}]```
And for django >= 4.2 (https://docs.djangoproject.com/en/4.2/releases/4.2/#logging) looks like:
```[{'sql': 'BEGIN', 'time': '0.000'}, {'sql': 'TRUNCATE RTINDEX `cluster`:`test___testapp_testmodel`', 'time': '0.000'}, {'sql': 'COMMIT', 'time': '0.000'}]```

Reply:
It's only what is logger. And what about really truncating table in transaction? This may be regression for 4.2 or this could be already there in 4.1, so no significant changes in 4.2 here. That's the question.

Reply:
@tumb1er it's really just a log entry
https://github.com/django/django/blob/stable/4.2.x/django/db/backends/base/base.py#L496
https://github.com/django/django/blob/stable/4.2.x/django/db/backends/base/base.py#L312
Transaction support is disabled. The flag feature `supports_transactions` is still here.
![изображение](https://github.com/just-work/django-manticore/assets/12427994/92b5a0f6-6acb-4839-ba1a-6b14d77320ea)

Reply:
first call `connection.autocommit(False)` and then `connection.commit()`